sars2seq/translate.py

```diff
@@ -94,8 +97,19 @@ def translateSpike(seq):
     current = 0
     seqLen = len(seq)
 
-    assert seqLen % 3 == 0, (f'The length of a sequence to be translated must '
-                             f'be a multiple of 3 but is {seqLen!r}.')
+    if not seqLen % 3 == 0:
+        raise TranslationError(
+            f'The length of a sequence to be translated must '
+            f'be a multiple of 3 but is {seqLen!r}.')
+
+    groups = groupby(seq)
+    result = [(label, sum([1 for _ in group])) for label, group in groups if
+              label == '-']
+
+    for gapStretch, count in result:
```

Comment:
I'd use `_` instead of `gapStretch` here, since the first thing is not used, and in any case will always be the `-` character.

Reply:
You could use `any` here:

```python
if any(length % 3 != 0 for _, length in result):
    # Error....
```

Reply:
Done